apps/crc_idle.py

```diff
@@ -71,14 +71,22 @@ def _count_idle_cpu_resources(cluster: str, partition: str) -> dict[int, dict[st
         """
 
         # Use `sinfo` command to determine the status of each node in the given partition
-        command = f'sinfo -h -M {cluster} -p {partition} -N -o %N,%C,%e'
+        command = f'sinfo -h -M {cluster} -p {partition} -N -o %N,%C,%e,%t'
         slurm_data = Shell.run_command(command).strip().split()
 
         # Count the number of nodes having a given number of idle cores/GPUs
         return_dict = dict()
         for node_info in slurm_data:
-            _, resource_data, free_mem = node_info.split(',')
-            allocated, idle, other, total = [int(x) for x in resource_data.split('/')]
+            _, resource_data, free_mem, node_state = node_info.split(',')
+
+            # If the node is in a downed state, report 0 resource availability.
+            down_keys = ['down', 'drain']
+            if any(re.search(key, node_state) for key in down_keys):
```

Comment:
Just for my own education, why `re.search` instead of `in`?

Reply:
I see. The `in` operator in Python can be used to check for substrings. so `"drain" in "drain*"` will return true.

```python
down_keys = ['down', 'drain']
if any(key in node_state for key in down_keys):
    ...
```

Regex might be more useful if you were looking for a complex pattern or had to check a very large number of values and wanted to do some clever optimization.